addons/io_hubs_addon/io/gltf_importer.py

```diff
@@ -1,342 +1,204 @@
-                        assign_property(gltf, blender_component, property_name, property_value)
-
-
-            except Exception:
-                print("Error encountered while adding Hubs components:")
-                traceback.print_exc()
-                print("Continuing on....\n")
-
-def create_scene_hubs_components(gltf):
-    if gltf.data.scene is None:
-        return
-
-    special_cases = {
-        'environment-settings': handle_environment_settings,
-        'background': handle_background,
-    }
-
-    gltf_scene = gltf.data.scenes[gltf.data.scene]
-    extensions = gltf_scene.extensions
-    if extensions:
-        MOZ_hubs_components = extensions.get('MOZ_hubs_components')
-        if MOZ_hubs_components:
-            for glb_component_name, glb_component_value in MOZ_hubs_components.items():
+def import_hubs_components(gltf_node, blender_object, import_settings):
+    if gltf_node and gltf_node.extensions and EXTENSION_NAME in gltf_node.extensions:
+        components_data = gltf_node.extensions[EXTENSION_NAME]
+        for component_name in components_data.keys():
+            component_class = get_component_by_name(component_name)
+            if component_class:
+                component_value = components_data[component_name]
                 try:
-                    if glb_component_name in special_cases.keys():
-                        special_cases[glb_component_name](gltf, glb_component_name, glb_component_value)
-                        continue
+                    component_class.gather_import(
+                        import_settings, blender_object, component_name, component_value)
+                except Exception as err:
+                    print(err)
+            else:
+                print('Could not import unsupported component "%s"' %
+                      (component_name))
 
-                    blender_component = add_hubs_component("scene", glb_component_name, glb_component_value, gltf=gltf)
 
-                    for property_name, property_value in glb_component_value.items():
-                        assign_property(gltf, blender_component, property_name, property_value)
+def add_lightmap(gltf_material, blender_mat, import_settings):
+    if gltf_material and gltf_material.extensions and 'MOZ_lightmap' in gltf_material.extensions:
+        extension = gltf_material.extensions['MOZ_lightmap']
 
-                except Exception:
-                    print("Error encountered while adding Hubs components:")
-                    traceback.print_exc()
-                    print("Continuing on....\n")
+        index = extension['index']
 
+        BlenderImage.create(
+            import_settings, index)
+        pyimg = import_settings.data.images[index]
+        blender_image_name = pyimg.blender_image_name
+        blender_image = bpy.data.images[blender_image_name]
+        blender_image.colorspace_settings.name = "Linear"
+        blender_mat.use_nodes = True
```

Comment:
Continue fix lightmap issues
```suggestion
        gltf_texture = import_settings.data.textures[texture_index]
        texture_extensions = gltf_texture.extensions
        if texture_extensions and texture_extensions.get('MOZ_texture_rgbe'):
            source = gltf_texture.extensions['MOZ_texture_rgbe']['source']
        else:
            source = gltf_texture.source

        BlenderImage.create(
            import_settings, source)
        pyimg = import_settings.data.images[source]
        blender_image_name = pyimg.blender_image_name
        blender_image = bpy.data.images[blender_image_name]
        if pyimg.mime_type == "image/vnd.radiance":
            blender_image.colorspace_settings.name = "Linear"
```